src/fides/api/alembic/migrations/versions/14acee6f5459_translation_data.py

```diff
@@ -0,0 +1,651 @@
+    load_default_experience_configs()
+)
+
+
+def backup_existing_data_csv(bind):
+    """
+    Perform a backup (dump to csv) of the data in our tables that will be impacted by migration.
+
+    Because this data migration is risky, it is recommended to perform a comprehensive DB backup/snapshot beforehand!
+    The "automated" dump to CSV here is an additional safety measure.
+    """
+    for table_name in [
+        "privacyexperience",
+        "privacyexperienceconfig",
+        "privacyexperienceconfighistory",
+        "privacynotice",
+        "privacynoticehistory",
+    ]:
+        dump_table_to_csv(bind, table_name)
+
+
+def remove_existing_experience_data(bind):
+    """
+    We remove any existing PrivacyExperience and PrivacyExperienceConfig records,
+    to allow for a "clean-slate" in creation of _new_ PrivacyExperienceConfigs based on the
+    "out of the box" templates.
+
+    This means that any existing user-edited data on PrivacyExperienceConfig records will need to be
+    _manually_ re-added to the post-migration state. It will not be migrated as part of the automatic migration.
+    """
+
+    def delete_experiences(bind):
+        """
+        Delete all PrivacyExperience records.
+
+        These records will be effectively re-created as a result of the "out of the box" creation
+        of 'new-style' PrivacyExperienceConfig records post-migration.
+        """
+        bind.execute(
+            text(
+                """
+                DELETE FROM privacyexperience;
+                """
+            )
+        )
+
+    def delete_experience_configs(bind):
+        """
+        Delete all PrivacyExperienceConfig records.
+
+        These records will be effectively re-created via the server's "out of the box" creation
+        of the updated PrivacyExperienceConfig records, post-migration.
+        """
+        bind.execute(
+            text(
+                """
+                DELETE FROM privacyexperienceconfig;
+                """
+            )
+        )
+
+    delete_experiences(bind)
+    delete_experience_configs(bind)
+
+
+def determine_needed_experience_configs(bind):
+    """
+    Our data model changes necessitate the creation of new Experience Configs. Now that "where" a Notice is served
+    is being moved from the "Notice" constructs to "Experiences", existing Experience Configs are not sufficient
+    to capture the full range of configuration on current Notices.
+    """
+
+    notice_query = """select id, name, regions, displayed_in_overlay, displayed_in_privacy_center, notice_key from privacynotice;"""
+    experience_config_query = """SELECT id, component, disabled, accept_button_label, banner_enabled, description, privacy_preferences_link_label, privacy_policy_link_label, privacy_policy_url, save_button_label, title, banner_description, banner_title, reject_button_label, acknowledge_button_label from privacyexperienceconfig;"""
```

Comment:
To add robustness here, just _in case_ another privacy experience config exists - we could filter these to ones whose ids match one of the ids in here? For example, if another overlay had been created, it is not repeatable which overlay's details get translated to the new US Overlay and EEA overlay.  This would force us to prioritize the updated details on the OOB experience configs over the custom ones.

https://github.com/ethyca/fides/blob/c91d46c9145891c150e03d39ff0c7f4b5bc37d56/src/fides/data/privacy_notices/privacy_experience_config_defaults.yml#L4

Reply:
yeah that's a good callout, i like it! so basically - ensure we only migrate experience configs we know about. any other experience configs will need to be handled manually.

Reply:
The addition of an OOB canadian config is a little interesting.  I assume updates to the original EEA overlay should be transferred there?

Reply:
yes i think that's right! and we should hardcode its regions i think? 🤷

Reply:
yes we could hardcode but we'd also want to make sure we didn't put canadian regions on the new EEA banner and modal (even if they are present on the original EEA overlay).  A manual migration step might be to call out which customers need to have the canadian config flipped on, depending on whether they had added canadian regions to their EEA overlay?